src/Microsoft.AspNetCore.OData/Extensions/ODataApplicationBuilderExtensions.cs

```diff
@@ -26,5 +37,50 @@ public static IApplicationBuilder UseODataBatching(this IApplicationBuilder app)
 
             return app.UseMiddleware<ODataBatchMiddleware>();
         }
+
+        /// <summary>
+        /// Use OData route with default route name and route prefix.
+        /// </summary>
+        /// <param name="app">The <see cref="IApplicationBuilder "/> to use.</param>
+        /// <param name="model">The <see cref="IEdmModel"/> to use.</param>
+        /// <returns>The <see cref="IApplicationBuilder "/>.</returns>
+        public static IApplicationBuilder UseOData(this IApplicationBuilder app, IEdmModel model)
+        {
+            return app.UseOData(DefaultRouteName, DefaultRoutePrefix, model);
+        }
+
+        /// <summary>
+        /// Use OData route with given route name and route prefix.
+        /// </summary>
+        /// <param name="app">The <see cref="IApplicationBuilder "/> to use.</param>
+        /// <param name="routeName">The given OData route name.</param>
+        /// <param name="routePrerix">The given OData route prefix.</param>
+        /// <param name="model">The <see cref="IEdmModel"/> to use.</param>
+        /// <returns>The <see cref="IApplicationBuilder "/>.</returns>
+        public static IApplicationBuilder UseOData(this IApplicationBuilder app, string routeName, string routePrerix, IEdmModel model)
+        {
+            if (app == null)
+            {
+                throw Error.ArgumentNull("app");
+            }
+
+            VerifyODataIsRegistered(app);
+
+            return app.UseMvc(b =>
+            {
+                b.Select().Expand().Filter().OrderBy().MaxTop(100).Count();
+
+                b.MapODataServiceRoute(routeName, routePrerix, model);
+            });
+        }
+
+        private static void VerifyODataIsRegistered(IApplicationBuilder app)
+        {
+            // We use the IPerRouteContainer to verify if AddOData() was called before calling UseOData
+            if (app.ApplicationServices.GetService(typeof(IPerRouteContainer)) == null)
+            {
+                throw Error.InvalidOperation(SRResources.MissingODataServices, "OData services");
```

Comment:
change to "nameof(IPerRouteContainer)"?

Reply:
As the comment (line 79) saying, I use "IPerRouteContainer" to test the OData services, not only the IPerRouteContainer. So, I input the "OData Services".

Reply:
I understand that the code is using IPerRouteCotnainer to check whether OData Services has been setup properly. But the usage of "OData Services" string is duplicated to what is already contained in MissingOdataServices, which already says 
```
"Cannot find '{0}'. OData services have not been configured. Are you missing a call to AddOData()?"
```

Reply:
+1, if we can use SRResources.MissingODataServices, that would be great

Reply:
#Changed.